src/task/SubscribeAndReceiveSubscribeAckTask.ts

```diff
@@ -94,6 +99,59 @@ export default class SubscribeAndReceiveSubscribeAckTask extends BaseTask {
     this.context.videoStreamIndex.integrateSubscribeAckFrame(subscribeAckFrame);
   }
 
+  // Our backends currently expect the video subscriptions passed in subscribe to precisely
+  // line up with the media sections, with a zero for any video send or inactive section.
+  //
+  // Firefox occasionally tosses stopped transceivers at the end of the SDP without reason
+  // and in general we don't want to be at the mercy of SDP sections not being in the same
+  // order as `getTransceivers`, so we simply recalculate the array here to enforce that
+  // expected invarient until we refactor our signaling to simply take a mapping of MID to
+  // subscription.
+  //
+  // This only works on Unified Plan SDPs
+  private fixUpSubscriptionOrder(sdp: string, videoSubscriptions: number[]): number[] {
+    if (this.context.transceiverController.getMidForStreamId === undefined) {
+      return videoSubscriptions;
+    }
+
+    const midsToStreamIds = new Map<string, number>();
+    for (const streamId of videoSubscriptions) {
+      // The local description will have been set by the time this task is running, so all
+      // of the transceivers should have `mid` set by now (see comment above `getMidForStreamId`)
+      const mid = this.context.transceiverController.getMidForStreamId(streamId);
+      if (mid === undefined) {
+        this.context.logger.warn(`Could not find MID for stream ID: ${streamId}`);
+        continue;
+      }
+      midsToStreamIds.set(mid, streamId);
+    }
+
+    const sections = new DefaultSDP(sdp).mediaSections();
```

Comment:
Wonder what's the point of interface if we hard coded SDP here :))

Reply:
The world may never know